dash-pipeline/bmv2/dash_outbound.p4

```diff
@@ -29,6 +30,41 @@ control outbound(inout headers_t hdr,
         meta.dropped = true;
     }
 
+    action route_service_tunnel(bit<1> is_overlay_dip_v4_or_v6,
+                                IPv4ORv6Address overlay_dip,
+                                bit<1> is_overlay_dip_mask_v4_or_v6,
+                                IPv4ORv6Address overlay_dip_mask,
+                                bit<1> is_overlay_sip_v4_or_v6,
+                                IPv4ORv6Address overlay_sip,
+                                bit<1> is_overlay_sip_mask_v4_or_v6,
+                                IPv4ORv6Address overlay_sip_mask,
+                                bit<1> is_underlay_dip_v4_or_v6,
+                                IPv4ORv6Address underlay_dip,
+                                bit<1> is_underlay_sip_v4_or_v6,
+                                IPv4ORv6Address underlay_sip,
+                                dash_encapsulation_t encap_type,
+                                bit<24> tunnel_id) {
+        /* Assume the overlay addresses provided are always IPv6 and the original are IPv4 */
+        assert(is_overlay_dip_v4_or_v6 == 1 && is_overlay_sip_v4_or_v6 == 1);
+        assert(is_overlay_dip_mask_v4_or_v6 == 1 && is_overlay_sip_mask_v4_or_v6 == 1);
+        assert(is_underlay_dip_v4_or_v6 != 1 && is_underlay_sip_v4_or_v6 != 1);
+        meta.encap_data.original_overlay_dip = hdr.ipv4.src_addr;
+        meta.encap_data.original_overlay_sip = hdr.ipv4.dst_addr;
+
+        service_tunnel_encode(hdr,
+                              overlay_dip,
+                              overlay_dip_mask,
+                              overlay_sip,
+                              overlay_sip_mask);
+
+        /* encapsulation will be done in apply block based on encap_type */
+        meta.encap_data.underlay_dip = underlay_dip == 0 ? meta.encap_data.original_overlay_dip : (IPv4Address)underlay_dip;
+        meta.encap_data.underlay_sip = underlay_sip == 0 ? meta.encap_data.original_overlay_sip : (IPv4Address)underlay_sip;
+        meta.encap_data.overlay_dmac = hdr.ethernet.dst_addr;
```

Comment:
Current assumption is to use original's packet dmac.

Reply:
Yes - it is copying the same value from original header